save-backend/src/main/kotlin/com/saveourtool/save/backend/security/VulnerabilityPermissionEvaluator.kt

```diff
@@ -0,0 +1,56 @@
+package com.saveourtool.save.backend.security
+
+import com.saveourtool.save.authservice.utils.AuthenticationDetails
+import com.saveourtool.save.backend.service.vulnerability.VulnerabilityService
+import com.saveourtool.save.backend.utils.hasRole
+import com.saveourtool.save.domain.Role
+
+import com.saveourtool.save.permission.Permission
+import com.saveourtool.save.utils.orNotFound
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Component
+
+/**
+ * Class that is capable of assessing user permissions.
+ */
+@Component
+class VulnerabilityPermissionEvaluator(
+    private val vulnerabilityService: VulnerabilityService,
+) {
+    /**
+     * Check permission for user to read, write and delete vulnerabilities by its [vulnerabilityName]
+     *
+     * @param authentication
+     * @param vulnerabilityName
+     * @param permission
+     * @return true if user with [authentication] has [permission] for [vulnerabilityName]
+     */
+    fun hasPermission(
```

Comment:
change the name to something more normal: `hasPermissionToEdit`

Reply:
I would leave as is because of `permission` param